jscomp/others/js_string.ml

```diff
@@ -76,6 +76,24 @@ external repeat : int -> t = "" [@@bs.send.pipe: t] (** ES2015 *)
 external replace : t ->  t ->  t = "" [@@bs.send.pipe: t]
 external replaceByRe : Js_re.t ->  t ->  t = "replace" [@@bs.send.pipe: t]
 
+(** returns a new string with some or all matches of a pattern replaced by the
+value returned from the given function
+
+@example {[
+let str = "hello world!"
+let re = [%re "/hello/g"]
+let replaced = Js.String.unsafeReplaceBy1 re (fun match p1 offset whole -> "hi")
+
+let () = Js.log replaced (* prints "hi world!" *)
+]}
+
+@see <https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter> MDN
+*)
```

Comment:
This looks really good, but could you add a copy of this to each function, and slightly tweak the example so it's appropriate for the function it's attached to? Ocamldoc associates a doc comment to just one specific item, the rest will not show any documentation at all in autocomplete and indexes and such.